File: [프로그래머스] 2018 KAKAO BLIND RECRUITMENT_[3차] 압축.py
def solution(msg):
    answer = []
    dic = {chr(i+ord("A")-1):i for i in range(1, 27)}
    idx = 27
    checkMsg = ""
    for i in msg:
        checkMsg += i
        if checkMsg not in dic:
            answer.append(dic[checkMsg[:-1]])
            dic[checkMsg] = idx
            idx += 1
            checkMsg = checkMsg[-1]
    answer.append(dic[checkMsg])
    return answer

File: [프로그래머스] 2018 KAKAO BLIND RECRUITMENT_[3차] 압축.js
function solution(msg) {
    var answer = [];
    let dic = {};
    for(let i = 1; i <= 26; i++){
        dic[String.fromCodePoint(64+i)] = i
    }
    let checkMsg = "";
    let idx = 27;
    for(let i of msg){
        checkMsg += i;
        if(Object.keys(dic).includes(checkMsg) === false){
            answer.push(dic[checkMsg.slice(0, checkMsg.length -1)]);
            dic[checkMsg] = idx
            idx += 1
            checkMsg = checkMsg[checkMsg.length-1];
        }
    }
    answer.push(dic[checkMsg]);
    return answer;
}